test/integration/cds_integration_test.cc

```diff
@@ -222,6 +222,123 @@ TEST_P(CdsIntegrationTest, CdsClusterTeardownWhileConnecting) {
   EXPECT_LE(cx_counter->value(), 1);
 }
 
+class LazyInitClusterStatsTest : public CdsIntegrationTest {
+
+public:
+  void initialize() {
+    this->use_real_stats_ = true;
+    config_helper_.addConfigModifier([](::envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      bootstrap.set_enable_lazyinit_stats(true);
+    });
+    CdsIntegrationTest::initialize();
+    test_server_->waitForCounterGe("cluster_manager.cluster_added", 1);
+  }
+
+  void SendRequestToCluster1AndWaitForResponse() {
+    BufferingStreamDecoderPtr response = IntegrationUtil::makeSingleRequest(
+        lookupPort("http"), "GET", "/cluster1", "", downstream_protocol_, version_, "foo.com");
+    ASSERT_TRUE(response->complete());
+    cleanupUpstreamAndDownstream();
+  };
+
+  void ExpectStats(absl::string_view cluster, bool initialized_gauge_exists, int initialized_gauge,
+                   bool cx_active_exists, int upstream_cx_active) {
+    if (initialized_gauge_exists) {
+      EXPECT_EQ(
+          test_server_->gauge(absl::StrCat("cluster.", cluster, ".ClusterTrafficStats.initialized"))
+              ->value(),
+          initialized_gauge);
+    } else {
+      EXPECT_EQ(test_server_->gauge(
+                    absl::StrCat("cluster.", cluster, ".ClusterTrafficStats.initialized")),
+                nullptr);
+    }
+    if (cx_active_exists) {
+      EXPECT_EQ(
+          test_server_->counter(absl::StrCat("cluster.", cluster, ".upstream_cx_total"))->value(),
+          upstream_cx_active);
+    } else {
+      EXPECT_EQ(test_server_->counter(absl::StrCat("cluster.", cluster, ".upstream_cx_total")),
+                nullptr);
+    }
+  }
+  void UpdateCluster1() {
```

Comment:
blank line above, and rename to updateCluster()

Reply:
done